rdflib/term.py

```diff
@@ -750,6 +762,35 @@ def __add__(self, val: Any) -> "Literal":
         if not isinstance(val, Literal):
             val = Literal(val)
 
+        # if self is datetime based and value is duration
+        if (
+            self.datatype in (_XSD_DATETIME, _XSD_DATE)
+            and val.datatype == _XSD_DURATION
+        ):
+            date1 = self.toPython()
+            duration = val.toPython()
```

Comment:
I'm slightly concerned about the data type here, I'm going to add some more tests to verify it, and some annotations to capture it. I think it can be `rdflib.term.Duration` or `datetime.timedelta` and I'm not sure it gets tested with `rdflib.term.Duration`.

Reply:
Sorry, I mean `isodate.Duration`: 

https://github.com/RDFLib/rdflib/blob/bceb6d778ea7f6bff57142e03bb479678b9186fd/rdflib/term.py#L1944

Reply:
Okay it does work with it, still will add some more tests around this tomorrow night.

Reply:
> Sorry, I mean `isodate.Duration`:

Eek. I was unaware of such a thing. Thanks for the catch.

Reply:
it has radd, and rsub (I think), so it works correctly